core/tx_pool.go

```diff
@@ -283,7 +283,6 @@ type TxPool struct {
 	signer      types.Signer
 	mu          sync.RWMutex
 	qiMu        sync.RWMutex
-	stateMu     sync.RWMutex
```

Comment:
no

Reply:
The statedb is never modified from the Txpool. It simply reads and sets the whole thing at once. This mutex in not needed but also creates a huge contention between the worker and the txpool.

Reply:
`pool.currentState` is a pointer that gets updated asynchronously.
`ProcessQiTx()` accepts the value in that pointer and reads from it later. Between the time that ProcessQiTx() is called and the time the state data is read, the value pointed to by `pool.currentState` may differ.